sklearn/tests/test_multioutput.py

```diff
@@ -491,3 +421,85 @@ def test_classifier_chain_vs_independent_models():
 
     assert_greater(jaccard_similarity_score(Y_test, Y_pred_chain),
                    jaccard_similarity_score(Y_test, Y_pred_ovr))
+
+
+def test_base_chain_fit_and_predict():
+    # Fit base chain and verify predict performance
+    X, Y = generate_multilabel_dataset_with_correlations()
+    chains = [RegressorChain(LinearRegression()),
+              ClassifierChain(LogisticRegression())]
+    for chain in chains:
+        chain.fit(X, Y)
+
+    for i in range(0, 2):
+        Y_pred = chains[i].predict(X)
+        assert_equal(Y_pred.shape, Y.shape)
+
+        assert_equal([c.coef_.size for c in chains[i].estimators_],
+                     list(range(X.shape[1], X.shape[1] + Y.shape[1])))
+
+    Y_prob = chains[1].predict_proba(X)
+    Y_binary = (Y_prob >= .5)
+    assert_array_equal(Y_binary, Y_pred)
+
+    assert isinstance(chains[0], RegressorMixin)
+    assert isinstance(chains[1], ClassifierMixin)
+
+
+def test_base_chain_fit_and_predict_with_sparse_data_and_cv():
+    # Fit base chain with sparse data cross_val_predict
+    X, Y = generate_multilabel_dataset_with_correlations()
+    X_sparse = sp.csr_matrix(X)
+    base_chain = [ClassifierChain(LogisticRegression(), cv=3),
+                  RegressorChain(LinearRegression(), cv=3)]
+    [chain.fit(X_sparse, Y) for chain in base_chain]
+    for i in range(0, 2):
+        Y_pred = base_chain[i].predict(X_sparse)
+        assert_equal(Y_pred.shape, Y.shape)
+
+
+def test_base_chain_random_order():
+    # Fit base chain with random order
+    X, Y = generate_multilabel_dataset_with_correlations()
+    for chain in [ClassifierChain(LogisticRegression()),
+                  RegressorChain(LinearRegression())]:
+        chain_random = clone(chain).set_params(order='random', random_state=42)
+        chain_random.fit(X, Y)
+        chain_fixed = clone(chain).set_params(order=chain_random.order_)
+        chain_fixed.fit(X, Y)
+        assert_not_equal(list(chain_random.order), list(range(4)))
```

Comment:
With pytest, we should prefer using the assert statement and comparison operators !=, ==, etc rather than these functions.